**Exploratory Data Analysis (EDA) Introduction**

Before diving into feature engineering or model training, it's essential to develop a thorough understanding of the dataset. This notebook focuses on exploratory data analysis (EDA) to identify patterns, spot anomalies, and uncover meaningful insights. The findings here will guide our decisions in the preprocessing, feature selection, and modeling phases of the project.

In [ ]:
import pandas as pd

df = pd.read_csv('../Data/WELFake_Dataset.csv')

if not df.empty:
    print('✅ Data read successfully')
    print(df.head())
else:
    print('❌ Data reading unsucessful')

**Dataset Overview**

In [14]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 72134 entries, 0 to 72133
Data columns (total 4 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   Unnamed: 0  72134 non-null  int64 
 1   title       71576 non-null  object
 2   text        72095 non-null  object
 3   label       72134 non-null  int64 
dtypes: int64(2), object(2)
memory usage: 2.2+ MB


In [15]:
df.describe(include = 'all')

Unnamed: 0                                              title  \
count   72134.000000                                              71576   
unique           NaN                                              62347   
top              NaN  Factbox: Trump fills top jobs for his administ...   
freq             NaN                                                 14   
mean    36066.500000                                                NaN   
std     20823.436496                                                NaN   
min         0.000000                                                NaN   
25%     18033.250000                                                NaN   
50%     36066.500000                                                NaN   
75%     54099.750000                                                NaN   
max     72133.000000                                                NaN   

         text         label  
count   72095  72134.000000  
unique  62718           NaN  
top                     NaN  
freq      738           NaN  
mean      NaN      0.514404  
std       NaN      0.499796  
min       NaN      0.000000  
25%       NaN      0.000000  
50%       NaN      1.000000  
75%       NaN      1.000000  
max       NaN      1.000000

In [16]:
df.isnull().sum()

Unnamed: 0      0
title         558
text           39
label           0
dtype: int64

These initial exploratory commands provide valuable insight into the structure and quality of the dataset. The dataset comprises **72,134 news articles**, each characterized by four variables: an unnamed index column (serving as an article ID), the article's title, the full text, and a binary label indicating whether the article is real or fake. Notably, **558** entries are missing titles and **39** are missing text. Addressing these missing values will be an essential part of the data preprocessing phase, and it is likely that records lacking core textual information will be removed to maintain the integrity of the analysis.